HW0/P4/P4.py

```diff
@@ -40,37 +49,124 @@
     # Read the observation array and plot it (Part 2)
     #####################
 
-    # ax.plot(x_coords, y_coords, z_coords,
-    #         '.g', label='Observed trajectory')
+    #Note: there is an easier way of doing this part, but I am using the longer way to practice
+    #my python =)
+
+    observation = np.loadtxt("P4_measurements.txt", delimiter=',')
+
+    #define the C matrix. only first 3 columns are needed
+    C = np.diag([1/rx, 1/ry, 1/rz])
+
+    #convert observation array into a matrix
+    ob_matrix = np.matrix(observation)
+
+    #create a matrix for the un-stretched values. Populated with zeros first
+    fixed_matrix = np.zeros(ob_matrix.shape)
+
+    #calculate the un-stretched values based on matrix multiplication of
+    #each row with the C matrix. Transpose the row to a column for the multiplication
+    #then transpose it back to a row, and save it to fixed_matrix
+    for x in range (0, ob_matrix.shape[0]):
+        fixed_matrix[x,:] = (C * ob_matrix[x,:].transpose()).transpose()
```

Comment:
It would have been faster to use matrix operation. For example, fixed_matrix = np.dot(C, ob_matrix.T)